### PR TITLE
Change dummy r dot java extra classpath to point to jars

### PR DESCRIPTION
**Pull request by kageiit**
Currently, buck project points the dummy r dot java output in the iml to the scratch directory. This is not always available especially when the outputs are materialized from the network cache. This changes the behavior to point to the final jar instead which will make sure the R classes can be found in all cases (from network, local build with both variations of `jar_spool_mode`).

This change is required to support Android Layout Preview in buck project and intellij.

### CODE REVIEW

**Comment by kageiit** [comment]
@christolliday any update on this?

File: src/com/facebook/buck/features/project/intellij/IjProjectTemplateDataPreparer.java [hunk 1]
```diff
@@ -493,6 +493,8 @@ private void addAndroidResourcePaths(
     if (resourcePaths.isEmpty()) {
       androidProperties.put(RESOURCES_RELATIVE_PATH_TEMPLATE_PARAMETER, EMPTY_STRING);
     } else {
+
```
Comment (artem-zinnatullin) [inline src/com/facebook/buck/features/project/intellij/IjProjectTemplateDataPreparer.java]: nit: unnecessary changes here?

Comment (kageiit) [reply src/com/facebook/buck/features/project/intellij/IjProjectTemplateDataPreparer.java]: god catch. I think I ran google java format on this and didnt need the changes after. ill remove

**Comment by christolliday** [comment]
Nice! please update the two integration tests and I'll merge

**Comment by christolliday** [comment]
Looks good thanks! Update the test data and I'll merge.

**Comment by kageiit** [comment]
@christolliday fixed one more test that travis spotted. Sry for the trouble. Should be good now
